### PR TITLE
Book search circulation dates filter #230

### PR DESCRIPTION
**Pull request by rlskoeser**
Adds circulation dates filter to book search and browse.

### CODE REVIEW

File: mep/books/models.py [hunk 1]
```diff
@@ -462,8 +462,13 @@ def index_data(self):
 
         earliest_date = self.earliest_date()
         if earliest_date:
+            # FIXME: acceptable for partial dates? (probably)
```
Comment (thatbudakguy) [inline mep/books/models.py:465]: this seems acceptable, or at least I'm not sure what the alternative would be.

File: mep/books/templates/books/work_list.html [hunk 2]
```diff
@@ -20,8 +20,13 @@
...
         {% include 'common/form_errors.html' %}
         <fieldset class="group">
-            {% comment %}
             <legend>Filter by:</legend>
+            <div class="inner">
+                <fieldset class="range facet {{ form.circulation_dates.css_classes }}" id="id_{{ form.circulation_dates.name }}">
+                    <legend>{{ form.circulation_dates.label }}</legend>
+                    {{ form.circulation_dates }}
+                </fieldset>
+            {% comment %}
```
Comment (thatbudakguy) [inline mep/books/templates/books/work_list.html:29]: it looks like there's an extra `<div class="inner">` here. doesn't seem to have affected the look of the page, but thought i would flag

File: mep/books/models.py (hunk 1 above)
Comment (rlskoeser) [reply mep/books/models.py:465]: Yeah, I think I already removed that fixme somewhere? (maybe in a different branch?) I think we're ok with unknowns sorting at the first, that's what they would do even with a partial date

File: mep/books/templates/books/work_list.html (hunk 2 above)
Comment (rlskoeser) [reply mep/books/templates/books/work_list.html:29]: whoops, thanks

File: mep/books/tests/test_books_views.py [hunk 3]
```diff
@@ -153,17 +154,32 @@ def test_get_queryset(self):
         view.request = self.factory.get(self.url)
         # if form is valid, should return all works sorted by chosen sort
         form.is_valid.return_value = True
-        form.cleaned_data = {'sort': 'title'}
+        form.cleaned_data = {'sort': 'title', 'circulation_dates': ''}
         solr_qs = view.get_queryset()
         db_qs = Work.objects.order_by('sort_title')
+        print(db_qs)
```
Comment (thatbudakguy) [inline mep/books/tests/test_books_views.py:160]: should this `print()` statement still be here?

File: mep/books/templates/books/work_list.html (hunk 2 above)
Comment (rlskoeser) [reply mep/books/templates/books/work_list.html:29]: @thatbudakguy this inner div mirrors what we have on the members search page (where I copied it from)

File: mep/books/tests/test_books_views.py (hunk 3 above)
Comment (rlskoeser) [reply mep/books/tests/test_books_views.py:160]: Nope! thank you for catching

**Pull request merged**